### ISSUE TITLE
UnicodeDecodeError not handled

### ISSUE THREAD

**Issue by lutoma** [garabik/grc#109]
`UnicodeDecodeError` is not handled and causes grc to crash. This is problematic for e.g. the `make` shim, where the commands run by make can output all sorts of things. It would probably be better to just pass invalid bytes through 1:1.

```python
Traceback (most recent call last):
  File "/usr/bin/grcat", line 180, in <module>
    line = freadline()
  File "/usr/lib/python3.7/codecs.py", line 322, in decode
    (result, consumed) = self._buffer_decode(data, self.errors, final)
UnicodeDecodeError: 'utf-8' codec can't decode byte 0xdf in position 0: invalid continuation byte
```

**Comment by cheese** [comment]
Met the same issue on Fedora 29 with grc 1.11.3

**Comment by perfgao** [comment]
Met the same issue on Mar 26 with grc 1.11.3

**Comment by Fukurokudzu** [comment]
Same here, grc crashes in debian with UnicodeDecodeError: 'utf-8' codec can't decode byte 0xfd in position 0: invalid start byte

**Comment by j-kwan** [comment]
Hi
Is there a workaround ? 
I have the same issue

**Comment by garabik** [comment]
Fixed in 1844de085c6e4ec8bd3e84611e61b8d6bb359aed - though you need at least python3.7 for a "proper" fix. There is a workaround for older pythons, but it makes the throughput slightly slower and might strip the characters.
